### ISSUE TITLE
Require assignments to match the nodes of the graph

### ISSUE THREAD

**Issue by maxhully** [mggg/GerryChain#280]
If the user tries to instantiate a Partition with an assignment whose keys don't match the nodes of the graph, we should raise an exception.

**Comment by zschutzman** [comment]
I think adding the following method to `Partition` 

```
    def validate_assignment(self):
	node_names = set(list(self.graph.nodes()))
	assgn_names = set([name for dist in self.assignment.parts.values() for name in dist])
	return node_names == assgn_names
```

and calling it in `Partition._first_time()`

```
    def _first_time(self, graph, assignment, updaters):
        self.graph = graph

        self.assignment = get_assignment(assignment, graph)

        if not self.validate_assignment():
             raise NameError("Graph's nodes' names do not match the Assignment's geo units' names.")

        if updaters is None:
            updaters = dict()
      ...
```

does the trick. I'm happy to do the pull request if you think this is a good way to solve it and this is the correct exception to raise.

**Comment by maxhully** [comment]
This is making me think that `Assignment` should implement the `Mapping` interface in [collections.abs](https://docs.python.org/3/library/collections.abc.html) (or at least have a `.keys()` method that iterates through all the names in the parts). If we do that, then I think we can inline `validate_assignment` to `set(assignment) == set(graph.nodes)`. For the error type, maybe `KeyError` or just plain old `ValueError` would be more appropriate?

How about you open a PR, and we continue this discussion there?

**Comment by maxhully** [comment]
Closed by #281 . Thanks Zach!
